doc/tutorial/erda.rst

```diff
@@ -0,0 +1,185 @@
+That is, the standard infrastructure to provide individual isolated Jupyter notebook containers to multiple users sharing a pool of actual compute nodes.
+
+.. figure:: /_images/erda/erda_jservice_homepage.png
+   :width: 100%
+   :align: center
+
+   Top fragment of Jupyter service home page.
+
+After clicking **Start My Server**, the site will give you an option to chose which notebook image you want to spawn.
+Select **HPC Notebook** as shown below and press the **Start** button.
+
+.. figure:: /_images/erda/erda_dag_image.png
+   :width: 100%
+   :align: center
+
+   Top fragment of Jupyter service home page with selected HPC Notebook image.
+
+This will spawn the **HPC Notebook** image and redirect you straight to the JupyterLab interface as shown below.
+The JupyterLab interface is the same in all available Services (DAG and MODI).
+
+.. figure:: /_images/erda/erda_dag_terminal.png
+   :width: 100%
+   :align: center
+
+   JupyterLab interface on DAG.
+
+Follow the Veros installation instructions below with respect to the selected service.
+
+Data Analysis Gateway (DAG)
++++++++++++++++++++++++++++
+
+In order to install Veros on a DAG instance do the following after launching the **Terminal**:
+
+1. Clone the Veros repository
+
+   .. exec::
+
+      from veros import __version__ as veros_version
+      if "+" in veros_version:
+          veros_version, _ = veros_version.split("+")
+      print(".. code-block::\n")
+      print(f"   $ git clone https://github.com/team-ocean/veros.git -b v{veros_version}")
+
+   (or `any other version of Veros <https://github.com/team-ocean/veros/releases>`__).
+
+2. Change the current directory to the Veros root directory::
+
+      $ cd veros
+
+3. Create a new conda environment for Veros, and install all relevant dependencies by running::
+
+      $ conda env create -f conda-environment.yml
+
+4. To use Veros, activate your new conda environment via::
+
+      $ conda activate veros
+
+5. Make a folder for your Veros setups, and switch to it::
+
+      $ mkdir ~/vs-setups
+      $ cd ~/vs-setups
+
+6. Copy the :doc:`global 4deg </reference/setups/4deg>` model template from the :doc:`setup gallery </reference/setup-gallery>`::
+
+      $ veros copy-setup global_4deg
+
+7. Change the current directory to the setup directory::
+
+      $ cd global_4deg/
+
+8. Modify model parameters with the `nano text editor <https://www.nano-editor.org>`__::
+
+      $ nano global_4deg.py
+
+9. Run the model in serial mode on one CPU core::
+
+      $ veros run global_4deg.py
+
+10. In case you want to run Veros in parallel mode, you need to reinstall the HDF5 library with parallel I/O support::
+
+      $ conda install "h5py=*=mpi_mpich*" --force-reinstall
+
+11. To run the model in parallel mode on 4 CPU cores execute::
+
+      $ mpirun -np 4 veros run global_4deg.py -n 2 2
```

Comment:
Instead of running in parallel I would rather use JAX (simpler to setup and better speed-ups)

Reply:
Isn't the global_4deg parallel run faster than JAX accelerated one? Anyways, I will check that to ensure the optimal performance.

Reply:
I think that JAX on 1 core will outperform NumPy on 4. Of course you can also do JAX on 4 cores :)

Reply:
**MPI:** 
 Current iteration: 100   (100.00/100.00d | 100.0% | 1.15m/(model year) | 0.0s left)
Integration done

Writing restart file 4deg_0100.restart.h5

Timing summary:
(excluding first iteration)
---
 setup time               = 1.63s
 main loop time           = 15.33s
   forcing                = 0.11s
   momentum               = 2.89s
     pressure             = 1.08s
     friction             = 1.11s
   thermodynamics         = 9.17s
   advection              = 0.25s
   EKE                    = 1.09s
   IDEMIX                 = 0.00s
   TKE                    = 1.24s
   boundary exchange      = 0.58s
 diagnostics and I/O      = 3.51s
 plugins                  = 0.00s

vs.

**JAX:**
 Current iteration: 100   (100.00/100.00d | 100.0% | 3.34m/(model year) | 0.0s left)
Integration done

Writing restart file 4deg_0100.restart.h5

Timing summary:
(excluding first iteration)
---
 setup time               = 15.99s
 main loop time           = 21.32s
   forcing                = 0.19s
   momentum               = 4.53s
     pressure             = 1.64s
     friction             = 1.56s
   thermodynamics         = 13.66s
   advection              = 0.38s
   EKE                    = 1.15s
   IDEMIX                 = 0.00s
   TKE                    = 1.16s
   boundary exchange      = 0.24s
 diagnostics and I/O      = 4.08s
 plugins                  = 0.00s

Reply:
OK, thanks for testing.

Reply:
You can't. There is some light thread parallelism in JAX that is not configurable. If you want true parallelism just use MPI + JAX.